.github/workflows/agora-docker.yml

```diff
@@ -0,0 +1,51 @@
+name: Build agora
+
+on: [push, pull_request]
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to DockerHub
+        if: github.event_name == 'push'
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Check out repository
+        uses: actions/checkout@v2
+        with:
+          submodules: false
+          persist-credentials: false
+
+      - name: Determining Agora version number
+        shell: bash
+        run: |
+          set -xeu
+          set -o pipefail
+          AGORA_VERSION=HEAD
+          GIT_DESCRIBE_OUT=`git describe --always`
+          if [ -n "$GIT_DESCRIBE_OUT" ]
+          then
+            AGORA_VERSION=$GIT_DESCRIBE_OUT
+          fi
+          echo "AGORA_VERSION=$AGORA_VERSION" >> $GITHUB_ENV
```

Comment:
I would check if `${{ github.ref }}` starts with  `refs/tags/` and uses that (minus the prefix), or use `${{ github.sha }}`. Or just use sha as a first step.

Reply:
https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#:~:text=Contexts%20are%20a%20way%20to,Contexts%20use%20the%20expression%20syntax.&text=Information%20about%20the%20workflow%20run,more%20information%2C%20see%20github%20context.

Reply:
I modified the code based on this, but the semantic of your suggested solution is different from mine. 

github.ref will only contain a tag if 'tag push' triggered the workflow, while in my solution git-describe will produce the latest tag regardless whether it was a 'tag push' or a 'branch push' or a PR.

Reply:
Perfect!